### PR TITLE
Add a variety of examples

### LINKED ISSUE #17: Add some examples (and perhaps operational considerations) especially for the Multi-CDN case
Some questions from implementers indicate that it would be helpful to both have examples
as well as operational considerations that makes it clear how to use this in a multiCDN scenario.
In-particular, examples showing how if there are multiple ECH keys supported by different sets of servers
then they need to have distinct TargetNames.

### PR DESCRIPTION
**Pull request by bemasc**
Fixes #17

### CODE REVIEW

File: draft-ietf-tls-svcb-ech.md [hunk 1]
```diff
@@ -90,6 +90,84 @@ Origins that publish an "ech" SvcParam in their HTTPS record SHOULD also publish
...
+@   3600 IN HTTPS 1 pool.cdn1.example. ech=ABC...
+            HTTPS 1 pool.cdn2.example. ech=DEF...
+; Apex IP records for compatibility with clients that do not support
+; HTTPS records.
+@   300  IN A    192.0.2.1
+            A    192.0.2.2
+            AAAA 2001:db8::1
+            AAAA 2001:db8::2
+
+www 3600  IN CNAME @
```
Comment (enygren) [inline draft-ietf-tls-svcb-ech.md]: What does a CNAME to @ mean here?  Following what we did in rfc9460:

```suggestion
; some times containing
www 3600  IN CNAME pool.cdn1.example.
; other times containing
www 3600  IN CNAME pool.cdn2.example.
```

File: draft-ietf-tls-svcb-ech.md [hunk 2]
```diff
@@ -90,6 +90,84 @@ Origins that publish an "ech" SvcParam in their HTTPS record SHOULD also publish
...
+$ORIGIN simple.example. ; Simple example zone
+@ 300 IN A     192.0.2.1
+         AAAA  2001:db8::1
+         HTTPS 1 . ech=ABC...
+www 300 IN A 192.0.2.1
+           AAAA 2001:db8::1
+           HTTPS 1 . ech=ABC...
+~~~
+{: title="Simple example zone with the same configuration on the apex and web domain."}
+
```
Comment (enygren) [inline draft-ietf-tls-svcb-ech.md:120]: Another example to add for Stephen Farrell's question might be different servers with different ECH values:

```suggestion

~~~
$ORIGIN heterogeneous.example. ; Example zone with two pools of servers
pool1 300 IN A     192.0.2.1
              AAAA  2001:db8:1::a
              AAAA  2001:db8:1::b
              AAAA  2001:db8:1::c
pool2 300 IN A     192.0.2.2
              AAAA  2001:db8:2::a
              AAAA  2001:db8:2::b
service         SVCB 1 pool1 ech=ABC...
service         SVCB 1 pool2 ech=DEF...
~~~
{: title="Service at service.heterogeneous.example. that round-robins between two server pools with different configurations.  Only supports clients which are SVCB-aware."}
```

This is perhaps the next "most simple" example that leads into the next ones.

**Review by enygren — COMMENTED** [review]
Thanks for writing these.  These otherwise LGTM.

File: draft-ietf-tls-svcb-ech.md (hunk 1 above)
Comment (bemasc) [reply draft-ietf-tls-svcb-ech.md]: CNAME to @ reuses the apex records, which saves 5 lines in this example.  Maybe I'm not understanding the question.

Note that this example is showing "merged" multi-CDN, as opposed to "selected" multi-CDN, which is what we showed in RFC 9460.

File: draft-ietf-tls-svcb-ech.md (hunk 2 above)
Comment (bemasc) [reply draft-ietf-tls-svcb-ech.md:120]: This example doesn't have any address records for `service`, so it's inaccessible to non-SVCB clients.

It's not clear to me what this is demonstrating that isn't clear from the multi-CDN example.

File: draft-ietf-tls-svcb-ech.md [hunk 3]
```diff
@@ -90,6 +90,84 @@ Origins that publish an "ech" SvcParam in their HTTPS record SHOULD also publish
...
+# Examples
+
+~~~
+$ORIGIN simple.example. ; Simple example zone
+@ 300 IN A     192.0.2.1
+         AAAA  2001:db8::1
+         HTTPS 1 . ech=ABC...
+www 300 IN A 192.0.2.1
+           AAAA 2001:db8::1
+           HTTPS 1 . ech=ABC...
```
Comment (paulwouters) [inline draft-ietf-tls-svcb-ech.md:102]: I thought one of the points of the HTTPS record was that you didn have to keep manually adding A/AAAA sync ups between the apex and www. of a zone ?

Comment (bemasc) [reply draft-ietf-tls-svcb-ech.md:102]: HTTPS records allow apex aliasing (as in the next example), but only for clients that support HTTPS records.  These examples mostly assume that compatibility with non-HTTPS-record-aware clients is required.

In this case, we could avoid duplicating the A/AAAA records by making `www` a CNAME to the apex (as in the multi-CDN example), but I thought that might not be the best choice for our simplest example.

**Comment by seanturner** [comment]
@bemasc @enygren merge when ready; see [email](https://mailarchive.ietf.org/arch/msg/tls/MHuPbJWXTUvXhIto1wSth3c2_Eo/).

File: draft-ietf-tls-svcb-ech.md [hunk 4]
```diff
@@ -90,6 +90,84 @@ Origins that publish an "ech" SvcParam in their HTTPS record SHOULD also publish
 
 If all HTTPS records for an alt-authority contain "ech" SvcParams, the client MUST adopt SVCB-reliant behavior (as in {{disabling-fallback}}) for that RRSet.  This precludes the use of certain connections that Alt-Svc would otherwise allow, as discussed in {{Section 9.3 of !SVCB}}.
 
+# Examples
+
+~~~
+$ORIGIN simple.example. ; Simple example zone
```
Comment (enygren) [inline draft-ietf-tls-svcb-ech.md:96]:
```suggestion
$ORIGIN simple.example. ; Simple example zone that retains compatibility for non-SVCB clients
```

File: draft-ietf-tls-svcb-ech.md (hunk 3 above)
Comment (enygren) [reply draft-ietf-tls-svcb-ech.md:102]: Perhaps we just be clear that this retains compatibility (see suggestion above),

File: draft-ietf-tls-svcb-ech.md (hunk 4 above)
Comment (enygren) [reply draft-ietf-tls-svcb-ech.md:96]: (or it might be better to put this in the title)

File: draft-ietf-tls-svcb-ech.md [hunk 5]
```diff
@@ -90,6 +90,84 @@ Origins that publish an "ech" SvcParam in their HTTPS record SHOULD also publish
...
+@ 300 IN A     192.0.2.1
+         AAAA  2001:db8::1
+         HTTPS 1 . ech=ABC...
+www 300 IN A 192.0.2.1
+           AAAA 2001:db8::1
+           HTTPS 1 . ech=ABC...
+~~~
+{: title="Simple example zone with the same configuration on the apex and web domain."}
```
Comment (enygren) [inline draft-ietf-tls-svcb-ech.md]:
```suggestion
{: title="Simple example zone with the same configuration on the apex and web domain and which continues to support clients which don't support SVCB."}
```

File: draft-ietf-tls-svcb-ech.md (hunk 2 above)
Comment (enygren) [reply draft-ietf-tls-svcb-ech.md:120]: Updated this to be clear that it only supports clients which support SVCB, and also switched the record type from HTTPS to SVCB as an example of using SVCB records.

This leads into the multi-CDN examples, but is outside the "CDN" context".  I'll also add some additional AAAA records above.  This was a case where an implementer was getting confused.

File: draft-ietf-tls-svcb-ech.md (hunk 1 above)
Comment (enygren) [reply draft-ietf-tls-svcb-ech.md]: I think it's much better to show a "selected" multi-CDN example.
(As a CDN vendor, I don't see it being operationally viable for CDN vendors to offer a merged multi-CDN implementation.  The automation required for that is out-of-scope from this draft and would be a huge operational pain.)
Trying to show the zone apex case may be adding more confusion than it helps with.

**Review by enygren — APPROVED** [review]

File: draft-ietf-tls-svcb-ech.md (hunk 1 above)
Comment (bemasc) [reply draft-ietf-tls-svcb-ech.md]: OK, I've incorporated these suggestions, with some additional tweaks.

**Review by MikeBishop — APPROVED** [review]

**Pull request merged**